src/movies.js

```diff
@@ -1,16 +1,132 @@
+    else{
+        let avrg;
+        let answer = movies.reduce((acc, elem)=>{
+            return acc + (elem.rate || 0);
+        },0)
+        avrg = answer/movies.length;
+        return parseFloat(avrg.toFixed(2));
+   }
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-
+function dramaMoviesRate(movies){
+    let dmSp;
+    let dramaMovies = movies.filter((movie)=>{
+        for(let i=0; i<movie.genre.length; i++){
+        dmSp = movie.genre[i] == "Drama";
+          if(dmSp){break;}
+        }
+        return dmSp;
+    })
+    return ratesAverage(dramaMovies);
+  
+}
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-
+function orderByYear(movies){
+    let cloneMovie = JSON.parse(JSON.stringify(movies));
+    let byYear = cloneMovie.sort((a,b)=>{
+        if(a.year>b.year){return 1;}
+        else if(a.year<b.year){return -1;}
+        else{
+            if(a.title>b.title){return 1;}
+            else if(a.title<b.title){return -1;}
+            else{return 0;}
+        }
+    });
+    return byYear;
+}
 // Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
+function orderAlphabetically(movies){
+    let cloneMovie = JSON.parse(JSON.stringify(movies));
+    let the20first=[];
+    let titlesInAlphabeticalOrder = cloneMovie.sort((a,b)=>{
+        if(a.title>b.title){return 1;}
+        else if(a.title<b.title){return -1;}
+        else{return 0;}
+    });
+    let arrOftitlesInAlphabeticalOrder = titlesInAlphabeticalOrder.map((elem,index)=>{
+      return elem.title;
+    });
+    the20first = arrOftitlesInAlphabeticalOrder.splice(0,20);
+    if(the20first<20){return arrOftitlesInAlphabeticalOrder;}
+    else{  return the20first;}
 
+}
 // BONUS - Iteration 7: Time Format - Turn duration of the movies from hours to minutes
+function turnHoursToMinutes(movies){
+    let cloneMovie = JSON.parse(JSON.stringify(movies));
+    let arrOfTheTime;
+    let hoursDuration;
+    let minutesDuration;
+    let filmDuration = cloneMovie.map((movie)=>{
+        if(movie.duration.includes(' ')){
+            arrOfTheTime = movie.duration.split(' '); //Now we have ['2h', '22min']
+            hoursDuration = arrOfTheTime[0].split('h')[0]; //
+            minutesDuration = arrOfTheTime[1].split('min')[0];
+            movie.duration = Number(hoursDuration) * 60 + Number(minutesDuration);
+        }
+        else if(movie.duration.includes('h')){
+            hoursDuration = movie.duration.split('h')[0];
+             movie.duration = Number(hoursDuration) * 60 
+        }
+        else{
+            minutesDuration = movie.duration.split('min')[0];
+            movie.duration = Number(minutesDuration);
+        }
+        return movie
+    });
+    return filmDuration;
+}
 
 // BONUS - Iteration 8: Best yearly rate average - Best yearly rate average
+function bestYearAvg(movies){
+    let moviesByYear = {}
+    if(movies ==0){return null;}
+    else{
```

Comment:
This works perfect but you don't really need to attach the rest of the code to an else. To make it more clean you can just create a simple condition if (movies.length === 0) without an else and then continue with the rest of the code. If the condition happens, then the return inside will stop the rest of the code in the function from executing.